spanner/transaction.go

```diff
@@ -1124,6 +1129,14 @@ func (t *ReadWriteTransaction) batchUpdateWithOptions(ctx context.Context, stmts
 	if err != nil {
 		return nil, err
 	}
+	// mark transaction and session to be eligible for long-running
+	t.mu.Lock()
+	t.isLongRunningTransaction = true
+	t.mu.Unlock()
+	t.sh.mu.Lock()
+	t.sh.isLongRunningTransaction = true
```

Comment:
Post a successful commit, session is returned back to the pool. In that case we need to reset these properties so that the state is not stale. For ex - If a session was being used with PDML, then these booleans will be true. But if the PDML is closed and next if it gets assigned to a read-only transaction, it should have these booleans as false. 

How is this being ensured?

Reply:
The `isLongRunningTransaction` property is set on `SessionHandle` object and not on the `Session` object. This `SessionHandle` is only linked for a transaction when it fetches a session from the pool, which means that we need not worry about resetting these properties.
https://github.com/googleapis/google-cloud-go/blob/f609b3cf831fb89c45386f81d0047560120cb3f4/spanner/session.go#L755-L757.

Post a successful commit, session is returned back to the pool which does not have the `isLongRunningTransaction` property.